src/main/java/io/confluent/connect/jdbc/sink/JdbcDbWriter.java

```diff
@@ -90,4 +80,7 @@ TableId destinationTable(String topic) {
     }
     return dbDialect.parseTableIdentifier(tableName);
   }
+
+  public void closeQuietly() {
```

Comment:
check if we are supposed to close the dataSource or the connections. Look into the `cachedConnectionProvider.close();` implementation and check what is happening.
You'll have to close the connections and throw a warning log in case of exceptions being raised to close them quietly and stopping the connector without any issues.